src/simdutf/icelake/simd32-inl.h

```diff
@@ -0,0 +1,56 @@
+template <typename T> struct simd32;
+
+template <> struct simd32<uint32_t> {
+  static const size_t SIZE = sizeof(__m512i);
+  static const size_t ELEMENTS = SIZE / sizeof(uint32_t);
+
+  __m512i value;
+
+  simdutf_really_inline simd32(const __m512i v) : value(v) {}
+
+  template <typename Pointer>
+  simdutf_really_inline simd32(const Pointer *ptr)
+      : value(_mm512_loadu_si512(reinterpret_cast<const __m512i *>(ptr))) {}
+
+  uint64_t sum() const {
+    const __m512i mask = _mm512_set1_epi64(0xffffffff);
+    const __m512i t0 = _mm512_and_si512(value, mask);
+    const __m512i t1 = _mm512_srli_epi64(value, 32);
+    const __m512i t2 = _mm512_add_epi64(t0, t1); // uint64_t
+
+    const __m256i lo = _mm512_extracti64x4_epi64(t2, 0);
```

Comment:
I think this sequence can be replaced with `_mm512_reduce_add_epi64` which looks [better in gcc](https://godbolt.org/z/85GKjz51Y).